portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,23 +44,37 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // initialize comments list and query
-    List<String> commentsList = new ArrayList<>();
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    
-    // call datastore query to retrieve Comment Entity's
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    // List<String> commentsList = new ArrayList<>();
+    try {
+        JSONArray commentList = new JSONArray();
+        JSONArray emailList = new JSONArray();
+        JSONObject j = new JSONObject();
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
-    //iterate through Comment Entity's retrieved and add comments to list
-    for (Entity entity : results.asIterable()) {
-      String commentFromData = (String) entity.getProperty("comment");
-      commentsList.add(commentFromData);
-    }
+        // call datastore query to retrieve Comment Entity's
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
 
-    // convert comments list to JSON and send it to main page
-    response.setContentType("/index.html;");
-    String json = new Gson().toJson(commentsList);
-    response.getWriter().println(json);
+        //iterate through Comment Entity's retrieved and add comments to list
+        for (Entity entity : results.asIterable()) {
+            String commentFromData = (String) entity.getProperty("comment");
+            String emailFromData = (String) entity.getProperty("email");
+            commentList.put(commentFromData);
+            emailList.put(emailFromData);
+            //   commentsList.add(commentFromData);
+        }
+
+        j.put("comments", commentList);
+        j.put("email", emailList);
+
+        // convert comments list to JSON and send it to main page
+        // response.setContentType("/index.html;");
+        // String json = new Gson().toJson(commentsList);
+        response.getWriter().println(j);
+     } catch (JSONException e) {
```

Comment:
You should only put the try-catch around the minimal number of lines necessary. This way, it's much easier to know what line of code is causing an exception when it does happen.

i.e. instead of:

```
try {
   <line of code that can throw exception>
   ... a bunch of other code...
} catch (...) {
   ...
}
```

It should be:
```
try {
   <line of code that can throw exception>
} catch (...) {
   ...
}
... a bunch of other code...
```

Reply:
i put all of the code in there because the lines of code that can throw and exception is when i create the JSONObject and JSONArray. if i take the remaining code out of the try catch, then the jsonobject and jsonarray is out of scope.